diff --git a/src/fountain.c b/src/fountain.c
index f5be392dd..faf54118e 100644
--- a/src/fountain.c
+++ b/src/fountain.c
@@ -67,7 +67,7 @@ dowaterdemon()
 
             /* Give those on low levels a (slightly) better chance of survival
              */
-            if (rnd(100) > (80 + level_difficulty())) {
+            if (rnd(110) > (80 + level_difficulty())) {
                 pline("Grateful for %s release, %s grants you a wish!",
                       mhis(mtmp), mhe(mtmp));
                 /* give a wish and discard the monster (mtmp set to null) */
@@ -230,7 +230,7 @@ drinkfountain()
 {
     /* What happens when you drink from a fountain? */
     register boolean mgkftn = (levl[u.ux][u.uy].blessedftn == 1);
-    register int fate = rnd(30);
+    register int fate = rnd(35);
 
     if (Levitation) {
         floating_above("fountain");
@@ -309,7 +309,7 @@ drinkfountain()
             exercise(A_CON, FALSE);
             /* this is more severe than rndcurse() */
             for (obj = g.invent; obj; obj = obj->nobj)
-                if (obj->oclass != COIN_CLASS && !obj->cursed && !rn2(5)) {
+                if (obj->oclass != COIN_CLASS && !obj->cursed && !rn2(7)) {
                     curse(obj);
                     ++buc_changed;
                 }
@@ -318,6 +318,7 @@ drinkfountain()
             break;
         }
         case 25: /* See invisible */
+        case 31:
             if (Blind) {
                 if (Invisible) {
                     You("feel transparent.");
@@ -334,6 +335,7 @@ drinkfountain()
             exercise(A_WIS, TRUE);
             break;
         case 26: /* See Monsters */
+        case 32:
             (void) monster_detect((struct obj *) 0, 0);
             exercise(A_WIS, TRUE);
             break;
diff --git a/src/mklev.c b/src/mklev.c
index 07feca909..ce6e41982 100644
--- a/src/mklev.c
+++ b/src/mklev.c
@@ -786,7 +786,7 @@ struct mkroom *croom;
         mkfount(0, croom);
     if (!rn2(60))
         mksink(croom);
-    if (!rn2(60))
+    if (!rn2(50))
         mkaltar(croom);
     x = 80 - (depth(&u.uz) * 2);
     if (x < 2)
@@ -1743,7 +1743,7 @@ struct mkroom *croom;
     /* Put a fountain at m.x, m.y */
     levl[m.x][m.y].typ = FOUNTAIN;
     /* Is it a "blessed" fountain? (affects drinking from fountain) */
-    if (!rn2(7))
+    if (!rn2(5))
         levl[m.x][m.y].blessedftn = 1;
 
     g.level.flags.nfountains++;
diff --git a/src/mkobj.c b/src/mkobj.c
index 7f97bae96..4108e5af6 100644
--- a/src/mkobj.c
+++ b/src/mkobj.c
@@ -30,23 +30,23 @@ static const struct icp mkobjprobs[] = { { 10, WEAPON_CLASS },
                                          { 10, ARMOR_CLASS },
                                          { 20, FOOD_CLASS },
                                          { 8, TOOL_CLASS },
-                                         { 8, GEM_CLASS },
+                                         { 6, GEM_CLASS },
                                          { 16, POTION_CLASS },
                                          { 16, SCROLL_CLASS },
                                          { 4, SPBOOK_CLASS },
                                          { 4, WAND_CLASS },
                                          { 3, RING_CLASS },
-                                         { 1, AMULET_CLASS } };
+                                         { 3, AMULET_CLASS } };
 
-static const struct icp boxiprobs[] = { { 18, GEM_CLASS },
+static const struct icp boxiprobs[] = { { 8, GEM_CLASS },
                                         { 15, FOOD_CLASS },
                                         { 18, POTION_CLASS },
-                                        { 18, SCROLL_CLASS },
-                                        { 12, SPBOOK_CLASS },
+                                        { 20, SCROLL_CLASS },
+                                        { 14, SPBOOK_CLASS },
                                         { 7, COIN_CLASS },
-                                        { 6, WAND_CLASS },
-                                        { 5, RING_CLASS },
-                                        { 1, AMULET_CLASS } };
+                                        { 8, WAND_CLASS },
+                                        { 7, RING_CLASS },
+                                        { 3, AMULET_CLASS } };
 
 static const struct icp rogueprobs[] = { { 12, WEAPON_CLASS },
                                          { 12, ARMOR_CLASS },
@@ -776,7 +776,7 @@ boolean artif;
         case WEAPON_CLASS:
             otmp->quan = is_multigen(otmp) ? (long) rn1(6, 6) : 1L;
             if (!rn2(11)) {
-                otmp->spe = rne(3);
+                otmp->spe = rne(3) + 1;
                 otmp->blessed = rn2(2);
             } else if (!rn2(10)) {
                 curse(otmp);
@@ -978,7 +978,7 @@ boolean artif;
                 otmp->spe = -rne(3);
             } else if (!rn2(10)) {
                 otmp->blessed = rn2(2);
-                otmp->spe = rne(3);
+                otmp->spe = rne(3) + 2;
             } else
                 blessorcurse(otmp, 10);
             if (artif && !rn2(40))
@@ -1359,7 +1359,7 @@ register int chance;
         return;
 
     if (!rn2(chance)) {
-        if (!rn2(2)) {
+        if (!rn2(3)) {
             curse(otmp);
         } else {
             bless(otmp);
diff --git a/src/pickup.c b/src/pickup.c
index ddabf097c..43ce54d68 100644
--- a/src/pickup.c
+++ b/src/pickup.c
@@ -2214,7 +2214,7 @@ int depthin;
 static boolean
 is_boh_item_gone()
 {
-    return (boolean) (!rn2(13));
+    return (boolean) (!rn2(20));
 }
 
 /* Scatter most of Bag of holding contents around.
diff --git a/src/pray.c b/src/pray.c
index f6ce19b1a..c825ccde5 100644
--- a/src/pray.c
+++ b/src/pray.c
@@ -1765,7 +1765,7 @@ dosacrifice()
             /* The player can gain an artifact */
             /* The chance goes down as the number of artifacts goes up */
             if (u.ulevel > 2 && u.uluck >= 0
-                && !rn2(10 + (2 * u.ugifts * nartifacts))) {
+                && !rn2(10 + (2 * u.ugifts + nartifacts))) {
                 otmp = mk_artifact((struct obj *) 0, a_align(u.ux, u.uy));
                 if (otmp) {
                     if (otmp->spe < 0)
diff --git a/src/timeout.c b/src/timeout.c
index 9782401b9..f98fabe5e 100644
--- a/src/timeout.c
+++ b/src/timeout.c
@@ -492,7 +492,7 @@ nh_timeout()
     boolean was_flying;
     int sleeptime;
     int m_idx;
-    int baseluck = (flags.moonphase == FULL_MOON) ? 1 : 0;
+    int baseluck = (flags.moonphase == FULL_MOON) ? 2 : 1;
 
     if (flags.friday13)
         baseluck -= 1;
diff --git a/src/trap.c b/src/trap.c
index 01c7a80fe..f5dc332ab 100644
--- a/src/trap.c
+++ b/src/trap.c
@@ -1271,7 +1271,7 @@ unsigned trflags;
                                 ? "stumbled into a pit of iron spikes"
                                 : "fell into a pit of iron spikes",
                        NO_KILLER_PREFIX);
-                if (!rn2(6))
+                if (!rn2(10))
                     poisoned("spikes", A_STR,
                              (conj_pit || adj_pit)
                                 ? "stepping on poison spikes"
diff --git a/sys/unix/hints/linux b/sys/unix/hints/linux
index 48ab631c1..a872cb974 100644
--- a/sys/unix/hints/linux
+++ b/sys/unix/hints/linux
@@ -10,7 +10,7 @@
 
 
 #PREFIX=/usr
-PREFIX=$(wildcard ~)/nh/install
+PREFIX=$(wildcard ~)/bin/nh
 HACKDIR=$(PREFIX)/games/lib/$(GAME)dir
 SHELLDIR = $(PREFIX)/games
 INSTDIR=$(HACKDIR)
